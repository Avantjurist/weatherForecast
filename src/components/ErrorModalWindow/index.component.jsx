import React, { Component } from 'react';
import * as styles from './index.css';

export default class ErrorModalWindow extends Component {
    handleClick = () => {
        const { cleanErrors, history } = this.props;

        cleanErrors();
        history.goBack();
    };

    render() {
        return (
            this.props.errors.length ?
                <div className={styles.popup}>
                    <div className={styles.popupInner}>
                        <div className={styles.errorMassage}>
                            {`${this.props.errors[this.props.errors.length - 1].status}
                             (${this.props.errors[this.props.errors.length - 1].statusText})`}
                        </div>
                        <button
                            className={styles.errorButton}
                            onClick={this.handleClick}
                        >
                            ok
                        </button>
                    </div>
                </div>
                : null
        );
    }
}
