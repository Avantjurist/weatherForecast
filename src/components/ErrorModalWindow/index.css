.popup {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    margin: auto;
    background-color: rgba(0,0,0, 0.5);
}

.popup-inner {
    position: absolute;
    width: 400px;
    height: 200px;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: auto;
    background: white;
    border-radius: 60px;
}

.error-massage {
    padding: 50px 20px;
    color: #d00d0d;
}

.error-button {
    width: 80px;
    height: 25px;
}