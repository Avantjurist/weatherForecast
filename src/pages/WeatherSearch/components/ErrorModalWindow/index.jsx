import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom'
import { cleanError } from '../../../../redux/actions/index';

class ErrorModalWindow extends Component {
    handleClick = () => {
        this.props.cleanError();
        this.props.history.goBack();
    };

    render() {
        return (
            this.props.errors.length ?
                <div>
                    <div>{this.props.errors}</div>
                    <button onClick={this.handleClick}>ok</button>
                </div> : null
        );
    }
}

const mapStateToProps = ({errors}) => ({errors});

const mapDispatchToProps = dispatch => ({
    cleanError: () => dispatch(cleanError()),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(ErrorModalWindow));
