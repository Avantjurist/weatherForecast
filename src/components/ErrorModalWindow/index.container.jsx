import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import ErrorModalWindow from './index.component';
import { cleanError } from '../../redux/actions/index';

const mapStateToProps = ({errors}) => ({errors});

const mapDispatchToProps = dispatch => ({
    cleanErrors: () => dispatch(cleanError()),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(ErrorModalWindow));
