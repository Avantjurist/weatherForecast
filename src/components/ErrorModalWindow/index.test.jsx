import React from 'react';
import { shallow } from 'enzyme';
import ErrorModalWindow from './index.component';

describe('Error modal window', () => {

    it('Render correctly ErrorModalWindow', () => {
        const props = { "errors": [] };
        const output = shallow(
            <ErrorModalWindow {...props} />
        );
        expect(output).toMatchSnapshot();
        expect(output.find("div")).toHaveLength(0);
        expect(output.find("popup")).toHaveLength(0);
    });

    it('Render correctly ErrorModalWindow with props', () => {
        const output = shallow(
            <ErrorModalWindow errors={[{ status: 'mock', statusText: 'status2', history: {} }]} />
        );
        expect(output).toMatchSnapshot();
    });

    it('call actions after click on the button', () => {
        const mockCleanErrors = jest.fn(), mockGoBack = jest.fn();
        const props = {
            "errors": [{ status: 'mock', statusText: 'status2' }],
            cleanErrors: mockCleanErrors,
            history: { goBack: mockGoBack }
        }
        const output = shallow(
            <ErrorModalWindow {...props} />
        );

        output.find('button').simulate('click', {
          })

        expect(mockCleanErrors).toHaveBeenCalledTimes(1);
        expect(mockCleanErrors).toHaveBeenCalledTimes(1);
    })
});
