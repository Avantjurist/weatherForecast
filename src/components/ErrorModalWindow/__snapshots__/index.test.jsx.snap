// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error modal window Render correctly ErrorModalWindow 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorModalWindow
    errors={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Error modal window Render correctly ErrorModalWindow with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorModalWindow
    errors={
      Array [
        Object {
          "history": Object {},
          "status": "mock",
          "statusText": "status2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
      >
        <div
          className={undefined}
        >
          mock
                             (status2)
        </div>
        <button
          className={undefined}
          onClick={[Function]}
        >
          ok
        </button>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
          >
            mock
                             (status2)
          </div>,
          <button
            className={undefined}
            onClick={[Function]}
          >
            ok
          </button>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mock
                             (status2)",
            "className": undefined,
          },
          "ref": null,
          "rendered": "mock
                             (status2)",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ok",
            "className": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "ok",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
        >
          <div
            className={undefined}
          >
            mock
                             (status2)
          </div>
          <button
            className={undefined}
            onClick={[Function]}
          >
            ok
          </button>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
            >
              mock
                             (status2)
            </div>,
            <button
              className={undefined}
              onClick={[Function]}
            >
              ok
            </button>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mock
                             (status2)",
              "className": undefined,
            },
            "ref": null,
            "rendered": "mock
                             (status2)",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ok",
              "className": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "ok",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
